app/controllers/tag_labels_controller.rb

```diff
@@ -0,0 +1,49 @@
+class TagLabelsController < ApplicationController
+  before_action :find_board, :find_tag
+  before_action :find_relation, only: :destroy
+
+  def create
+    label_ids = params[:label]
+    label_ids.each do |item|
+      new_relation = TagLabel.new label_id: item, tag_id: params[:tag_id]
+      flash[:danger] = t".fail" unless new_relation.save
```

Comment:
chỗ này bị n+1 rồi bạn, sửa lại cho hiệu suất tốt hơn nehs

Reply:
vang

Reply:
ở đây em lấy 1 mảng id từ bên view để add vào bảng trung gian để tạo quan hệ anh ạ nên ở SQL thì sẽ chạy nhiều lệnh INSERT_INTO như bình thường vẫn insert ạ, và em phải dùng foreach để lấy được từng phần tử trong mảng để insert ạ

Reply:
1 cái save thất bại thì sao e ?
e cho nó vào transaction nhé

Reply:
1 cái save thất bại thì em cho nó chạy tiếp để còn save những cái label có thể add ạ, nếu cho vào transaction thì em nghĩ nó sẽ hủy hết tất cả những label em add ạ, em có xử lý trường hợp thất bại thì sẽ in ra thông báo bao nhiêu label không được add rồi ạ